app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,56 @@
+class PasswordResetsController < ApplicationController
+  before_action %i(get_user valid_user check_expiration), only: %i(edit update)
+
+  def new; end
+
+  def create
+    @user = User.find_by email: params[:password_reset][:email].downcase
+    if @user
+      @user.create_reset_digest
+      @user.send_password_reset_email
+      flash[:info] = t "reset_email_sent"
+      redirect_to root_url
+    else
+      flash.now[:danger] = t "email_not_found"
+      render :new
+    end
+  end
+
+  def edit; end
+
+  def update
+    if params[:user][:password].blank?
+      @user.errors.add :password, t("not_empty")
+      render :edit
+    elsif @user.update user_params
+      log_in @user
+      @user.update_attribute :reset_digest, nil
+      flash[:success] = t "password_reset.done"
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  private
+  def user_params
+    params.require(:user).permit :password, :password_confirmation
+  end
+
+  def get_user
+    @user = User.find_by email: params[:email]
+  end
+
+  def valid_user
+    return if @user&.activated? AND @user.authenticated?(:reset, params[:id])
```

Comment:
```suggestion
    return if @user&.activated? && @user.authenticated?(:reset, params[:id])
```

Reply:
@huongnt-2545 cho em hỏi về cái này nếu em để && nó sẽ báo lỗi Ensure that safe navigation is used consistently inside of && and ||. nêú em đổi thành @user&.activated?&@user.authenticated?(:reset, params[:id]) nó vẫn báo lỗi rubocop còn để AND thì bình thường ạ :(, em lên mạng tra nó bảo báo lỗi như chị viết là do bug của rails nhưng em ko chắc lắm nên cứ sửa tạm thành AND

Reply:
@user && @user.activated? && @user.authenticated?(:reset, params[:id])

Reply:
@anlvq-1058  thế rubocop nó lai bắt gộp 2 thằng đầu vào 1 mà anh?